src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1784,6 +1784,10 @@ private void OnPreferencesWindowClick(object sender, RoutedEventArgs e)
         {
             var preferencesWindow = new PreferencesView(this);
             preferencesWindow.WindowStartupLocation = WindowStartupLocation.CenterOwner;
+            // when the preferences window is opened update the package paths UI based on most up to date preferences model.
+            // TODO refactor and generalize this solution after 2.12 release - what about other preferences
+            // that can be updated on the model (from an extension for example) and need to be updated in the UI?
+            (preferencesWindow.PackagePathView.DataContext as PackagePathViewModel)?.InitializeRootLocations();
```

Comment:
Can we raise an event here that is handled by `PackagePathViewModel`? That way `DynamoView `does not need to know about `PackagePathViewModel`.

Reply:
@aparajit-pratap totally, let me give that a shot.

Reply:
@aparajit-pratap ran into some trouble doing using events, came up with another 1 liner that I think is cleaner than this - if you think an event is a better solution, let's talk, can show you the issues I'm running into, -mostly about cleaning up the event subscription.